attachment_queue/models/attachment_queue.py

```diff
@@ -0,0 +1,105 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+import logging
+
+from odoo import api, fields, models, registry
+
+_logger = logging.getLogger(__name__)
+
+
+class AttachmentQueue(models.Model):
+    _name = "attachment.queue"
+    _inherits = {"ir.attachment": "attachment_id"}
+    _inherit = ["mail.thread"]
+
+    attachment_id = fields.Many2one(
+        "ir.attachment",
+        required=True,
+        ondelete="cascade",
+        help="Link to ir.attachment model ",
+    )
+    file_type = fields.Selection(
+        selection=[],
+        help="The file type determines an import method to be used "
+        "to parse and transform data before their import in ERP or an export",
+    )
+    date_done = fields.Datetime()
+    state = fields.Selection(
+        [("pending", "Pending"), ("failed", "Failed"), ("done", "Done")],
+        readonly=False,
+        required=True,
+        default="pending",
+    )
+    state_message = fields.Text()
+    failure_emails = fields.Char(
+        compute="_compute_failure_emails",
+        string="Failure Emails",
+        help="Comma-separated list of email addresses to be notified in case of"
+        "failure",
+    )
+
+    def _compute_failure_emails(self):
+        for attach in self:
+            attach.failure_emails = attach._get_failure_emails()
+
+    def _get_failure_emails(self):
+        # to be overriden in submodules implementing the file_type
+        self.ensure_one()
+        return ""
+
+    @api.model
+    def run_attachment_queue_scheduler(self, domain=None):
+        if domain is None:
+            domain = [("state", "=", "pending")]
+        batch_limit = self.env.ref(
+            "attachment_queue.attachment_queue_cron_batch_limit"
+        ).value
+        if batch_limit and batch_limit.isdigit():
+            limit = int(batch_limit)
+        else:
+            limit = 200
+        attachments = self.search(domain, limit=limit)
+        if attachments:
+            return attachments.run()
+        return True
+
+    def run(self):
+        """
+        Run the process for each attachment queue
+        """
+        failure_tmpl = self.env.ref("attachment_queue.attachment_failure_notification")
+        for attachment in self:
+            with api.Environment.manage():
+                with registry(self.env.cr.dbname).cursor() as new_cr:
+                    new_env = api.Environment(new_cr, self.env.uid, self.env.context)
+                    attach = attachment.with_env(new_env)
+                    try:
+                        attach._run()
+                    # pylint: disable=broad-except
+                    except Exception as e:
+                        attach.env.cr.rollback()
+                        _logger.exception(str(e))
+                        attach.write({"state": "failed", "state_message": str(e)})
```

Comment:
RFC Off topic of the migration, concern improvement about retry run task
If state_message is `could not serialize access due to concurrent update` could we consider to let state to 'pending'
Then there'll be a retry to the next execution job without do nothing else.

Reply:
Yes I guess it could be nice. But then, it would be nice to have a max retry number, it could be nice to set a delay before retry, etc... 
All of this is already manage by queue job, I wonder if it is worse it to try to re-implement all of this here.

Did you already get a lot of case of conccurent update using this module in production ?
In my cases, I don't think I do.

Working to integrate the jobs here could be better than reinvent the wheel, also because other queue job features may appears as nice in the future (the priority management for instance)...but it is not as simple as it may seem.

Reply:
Thanks a lot for your feedback. In my experience with EDI I have some examples.
Converge with jobs is an option, yes.

Reply:
Pros : all is already in queue.job

Cons : security with jobs doesn't match with end users

Reply:
integrating queue.job into this would mean rethinking the module structure
It would be interesting to know what the context is to trigger "could not serialize access due to concurrent update" error

Reply:
Finally, I think that in case queue is installed, a job could be created to manage the task execution. No need to have a dependency on it. What do you think ?

Reply:
How about another module attachment_synchronize_async ?